src/main/java/dao/TicketDao.java

```diff
@@ -0,0 +1,86 @@
+package dao;
+
+import jdbc.CustomDataSource;
+import ticket.Ticket;
+import ticket.TicketType;
+
+import java.sql.*;
+import java.util.ArrayList;
+import java.util.List;
+
+public class TicketDao {
+    private final CustomDataSource dataSource = CustomDataSource.getInstance();
+    private Connection connection = null;
+    private PreparedStatement preparedStatement = null;
+
+    private static final String SAVE_TICKET = "INSERT INTO tickets (ticket_id, user_id, ticket_type, creation_date) VALUES (?, ?, ?::ticket_type, ?)";
+    private static final String FETCH_TICKET_BY_ID = "SELECT ticket_id, user_id, ticket_type, creation_date FROM tickets WHERE ticket_id = ?";
+    private static final String FETCH_TICKET_BY_USER_ID = "SELECT ticket_id, user_id, ticket_type, creation_date FROM tickets WHERE user_id = ?";
+    private static final String UPDATE_TICKET_TYPE = "UPDATE tickets SET ticket_type = ?::ticket_type WHERE id = ?";
+
+    public void saveTicket(Ticket ticket, int userId) {
+        try {
+            connection = dataSource.getConnection();
+            preparedStatement = connection.prepareStatement(SAVE_TICKET);
+            preparedStatement.setInt(1, ticket.getId());
+            preparedStatement.setInt(2, userId);
+            preparedStatement.setString(3, ticket.getTicketType().name());
+            preparedStatement.setTimestamp(4, Timestamp.valueOf(ticket.getTicketCreationTime()));
+            System.out.println(preparedStatement.executeUpdate());
+            preparedStatement.close();
+            connection.close();
```

Comment:
close resources in `finally` block or use `try-with-resources` construction

Reply:
Fix in https://github.com/kolosokvit/ticket-service/pull/4/commits/5795506ba09e5903e2930c8aadd7d78b81441ff9